apps/trading/e2e/actions/utils.py

```diff
@@ -46,3 +48,19 @@ def change_keys(page: Page, vega:VegaServiceNull, key_name):
     page.get_by_test_id("key-" + vega.wallet.public_key(key_name)).click()
     page.click(f'data-testid=key-{vega.wallet.public_key(key_name)} >> .inline-flex')
     page.reload()
+
+def forward_time(vega: VegaService, forward_epoch: bool = False):
+    vega.wait_fn(1)
+    vega.wait_for_total_catchup()
+
+    if forward_epoch:
+        next_epoch(vega)
+
+def selector_contains_text(page: Page, selector, expected_text, timeout=5000):
```

Comment:
Why aren't we just using the playwright expect?

Reply:
remember when we spoke about the initial values being incorrect? It fails because it finds the element, and the element initially has the incorrect text so fails the assertion. This ensures it gives the text chance to update.

Reply:
ahh, fuck that's annoying. I remember.